core/application/src/app.rs

```diff
@@ -74,8 +77,8 @@ impl<C: Collection> ApplicationInterface<C> for Application<C> {
     /// # Safety
     ///
     /// See the safety document for the [`ExecutionEngineSocket`].
-    fn transaction_executor(&self) -> ExecutionEngineSocket {
-        self.update_socket.clone()
+    fn transaction_executor(&self) -> Option<ExecutionEngineSocket> {
```

Comment:
```suggestion
    fn transaction_executor(&self) -> ExecutionEngineSocket> {
```
Lets keep the return on the interface the same as it was. We should unwrap here in this function before returning. The safety comments on this function in the interface should reflect that this is going to panic if a process besides consensus calls this